rllib/utils/space_utils.py

```diff
@@ -0,0 +1,53 @@
+from gym.spaces import Tuple, Dict
+
+
+def flatten_space(space):
+    """Flattens a gym.Space into its primitive components.
+
+    Primitive components are any non Tuple/Dict spaces.
+
+    Args:
+        space(gym.Space): The gym.Space to flatten. This may be any
+            supported type (including nested Tuples and Dicts).
+
+    Returns:
+        List[gym.Space]: The flattened list of primitive Spaces. This list
```

Comment:
This should be a super flat space gym.Box() right?

Reply:
Removed Beta for now. Will be added in a separate PR.

Reply:
Nope, this corresponds to tree.flatten, which returns a list. The problem is that tree.flatten does not work on gym.Dict/Tuple, so I had to add this helper here. What you mean is "super-flatten", where everything really gets crunched into a single tensor. I should move our function for that purpose (currently called `_flatten_action` in episode.py) into space_utils as well and rename it "flatten_to_single_tensor" for clarity.

Reply:
Ok, I guess I'm a bit confused why we need this intermediate "list of gym spaces", but maybe the next PR will make it clear.